client/core/core_test.go

```diff
@@ -1432,50 +1445,52 @@ func TestBookFeed(t *testing.T) {
 }
 
 type tDriver struct {
-	f           func(*asset.WalletConfig, dex.Logger, dex.Network) (asset.Wallet, error)
-	decoder     func(coinID []byte) (string, error)
-	winfo       *asset.WalletInfo
-	doesntExist bool
-	existsErr   error
-	createErr   error
-}
-
-func (drv *tDriver) Exists(walletType, dataDir string, settings map[string]string, net dex.Network) (bool, error) {
-	return !drv.doesntExist, drv.existsErr
-}
-
-func (drv *tDriver) Create(*asset.CreateWalletParams) error {
-	return drv.createErr
+	wallet        asset.Wallet
+	decodedCoinID string
+	winfo         *asset.WalletInfo
 }
 
 func (drv *tDriver) Open(cfg *asset.WalletConfig, logger dex.Logger, net dex.Network) (asset.Wallet, error) {
-	return drv.f(cfg, logger, net)
+	return drv.wallet, nil
 }
 
 func (drv *tDriver) DecodeCoinID(coinID []byte) (string, error) {
-	return drv.decoder(coinID)
+	return drv.decodedCoinID, nil
 }
 
 func (drv *tDriver) Info() *asset.WalletInfo {
 	return drv.winfo
 }
 
+type tCreator struct {
+	*tDriver
+	doesntExist bool
+	existsErr   error
+	createErr   error
+}
+
+func (ctr *tCreator) Exists(walletType, dataDir string, settings map[string]string, net dex.Network) (bool, error) {
+	return !ctr.doesntExist, ctr.existsErr
+}
+
+func (ctr *tCreator) Create(*asset.CreateWalletParams) error {
+	return ctr.createErr
+}
+
 func TestCreateWallet(t *testing.T) {
 	rig := newTestRig()
 	defer rig.shutdown()
 	tCore := rig.core
 
 	// Create a new asset.
-	a := *tDCR
+	a := *tUTXOAssetA
 	tILT := &a
 	tILT.Symbol = "ilt"
 	tILT.ID, _ = dex.BipSymbolID(tILT.Symbol)
 
 	// ILT is based on DCR, so use it's coinID decoder. If this changes, update
 	// the following decoder function:
-	decoder := func(coinID []byte) (string, error) {
-		return asset.DecodeCoinID(tDCR.ID, coinID) // using DCR decoder
-	}
+	decodedCoinID := "dcr"
```

Comment:
`"ilt"` then?  Also the comment seems to be mostly obsolete now, at least the part about the function, but I don't see the reason to keep the `"dcr"` decoding either.

And does this need to be a variable at all or can we just set this directly in `asset.Register(tILT.ID, , &tDriver{... ` below?

Reply:
Yes.